src/ray/raylet/raylet_client.h

```diff
@@ -0,0 +1,165 @@
+#ifndef RAYLET_CLIENT_H
+#define RAYLET_CLIENT_H
+
+#include <unistd.h>
+#include <mutex>
+
+#include "ray/raylet/task_spec.h"
+#include "ray/status.h"
+
+using ray::ActorID;
+using ray::JobID;
+using ray::ObjectID;
+using ray::TaskID;
+using ray::UniqueID;
+
+using MessageType = ray::protocol::MessageType;
+
+class RayletConnection {
```

Comment:
We probably don't want this class. Instead, we should just use the `ClientConnection` class that we already have.

@pcmoritz can you comment?

Reply:
However, here's some issues:
1. We need to keep locks.
2. I do not know how to establish a connection for socket path with `ClientConnection`.

Reply:
Yeah, it should be possible to make these work. Let's look into this together tomorrow :)

Reply:
Converting to `ClientConnection` is great, but may require large changes. Maybe it's okay to do that in a different PR?